src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -0,0 +1,478 @@
+      ...item,
+      isSelected: selectedIds.has(item.entityId),
+      setSelected: (value: boolean) => {
+        return value
+          ? addSelectedId(item.entityId)
+          : removeSelectedId(item.entityId)
+      },
+    }
+  })
+
+  function addItemsToDataset(newItems: Reference[]) {
+    setDatasetToUpdate(datasetToUpdate => {
+      if (datasetToUpdate) {
+        const existingItems = datasetToUpdate.items.filter(
+          item => !newItems.find(newItem => newItem.targetId === item.entityId),
+        )
+
+        if (existingItems.length < datasetToUpdate.items.length) {
+          displayToast(
+            'Files were added that were already in the dataset. The versions of those files might have been updated.',
+            'info',
+          )
+        }
+
+        const items = [
+          ...existingItems,
+          ...newItems.map(item => ({
+            entityId: item.targetId,
+            versionNumber: item.targetVersionNumber!,
+          })),
+        ]
+        return {
+          ...datasetToUpdate,
+          items: items,
+        }
+      } else {
+        console.warn(
+          'Cannot add items to the Dataset because is undefined. The Dataset may not have been fetched yet.',
+        )
+        return datasetToUpdate
+      }
+    })
+    clearSelectedIds()
+  }
+
+  function removeSelectedItemsFromDataset() {
+    setDatasetToUpdate(dataset => ({
+      ...dataset!,
+      items: dataset!.items.filter(
+        datasetItem => !selectedIds.has(datasetItem.entityId),
+      ),
+    }))
+
+    clearSelectedIds()
+  }
+
+  function changeVersionOnItem(entityId: string, newVersion: number) {
+    setDatasetToUpdate(dataset => ({
+      ...dataset!,
+      items: dataset!.items.map(datasetItem =>
+        datasetItem.entityId === entityId
+          ? { entityId: entityId, versionNumber: newVersion }
+          : datasetItem,
+      ),
+    }))
+  }
+
+  const DatasetItemVersionRenderer = (
+    props: CellRendererProps<DatasetItemsEditorTableData>,
+  ) => {
+    return (
+      <DatasetEditorVersionRenderer
+        {...props}
+        toggleSelection={datasetItem => {
+          changeVersionOnItem(datasetItem.entityId, datasetItem.versionNumber)
+        }}
+      />
+    )
+  }
+
+  type SelectAllCheckboxRendererProps = {
+    datasetToUpdate: Dataset
+    selectedIds: Omit<Set<string>, 'add' | 'delete' | 'clear'>
+    addSelectedId: (...items: string[]) => void
+    clearSelectedIds: () => void
+    allItemsAreSelected: boolean
+  }
+  const SelectAllCheckboxRenderer = (props: SelectAllCheckboxRendererProps) => {
+    const { datasetToUpdate, clearSelectedIds, addSelectedId } = props
+    useTraceUpdate(props)
+    const isChecked = allItemsAreSelected
+
+    return datasetToUpdate ? (
+      <div
+        data-testid="Select All"
+        style={{ cursor: 'pointer' }}
+        onClick={() => {
+          if (isChecked) {
+            clearSelectedIds()
+          } else if (datasetToUpdate) {
```

Comment:
This "if (datasetToUpdate)" is unnecessary, right (we only render this block if datasetToUpdate)?